runtime/server_http_response.go

```diff
@@ -110,26 +110,30 @@ func (res *ServerHTTPResponse) SendErrorString(
 		zap.String("path", res.Request.URL.Path),
 	)
 
-	res.WriteJSONBytes(statusCode,
+	res.WriteJSONBytes(statusCode, nil,
 		[]byte(`{"error":"`+err+`"}`),
 	)
 }
 
 // WriteJSONBytes writes a byte[] slice that is valid json to Response
 func (res *ServerHTTPResponse) WriteJSONBytes(
-	statusCode int, bytes []byte,
+	statusCode int, headers map[string]string, bytes []byte,
 ) {
 	// TODO: mark header as pending ?
 	res.responseWriter.Header().
 		Set("content-type", "application/json")
 
+	for k, v := range headers {
```

Comment:
Let's set headers first and then finally set the content-type.

We do not want application headers to overwrite the content-type.